addons/mail/static/tests/message/message_reply.test.js

```diff
@@ -120,3 +121,28 @@ test("click on message in reply highlights original message", async () => {
     );
     await contains(".o-mail-Message.o-highlighted:contains('This message has been removed')");
 });
+
+test("reply action functionality in Chatter", async () => {
+    const pyEnv = await startServer();
+    const partnerBId = pyEnv["res.partner"].create({ name: "Partner B" });
+    pyEnv["mail.message"].create({
+        body: "Test message from B",
+        subtype_id: pyEnv["mail.message.subtype"].search([
+            ["subtype_xmlid", "=", "mail.mt_note"],
+        ])[0],
+        model: "res.partner",
+        res_id: serverState.partnerId,
+        author_id: partnerBId,
+    });
+    await start();
+    await openFormView("res.partner", serverState.partnerId);
+    await click(".o-mail-Message-actions button[title='Reply']");
+    await contains("button.active", { text: "Log note" });
+    await contains(".o-mail-Composer-coreHeader", { text: "Replying to Partner B", count: 0 });
+    await contains(".o-mail-Composer-input:focus", { value: "@Partner B " });
+    await click(".o-mail-Composer-send");
+    await contains(".o-mail-Message a.o_mail_redirect", { text: "@Partner B" });
+    await click(".o-mail-Message.o-selfAuthored button[title='Expand']");
+    await contains(".o-dropdown-item:contains('Forward')");
+    await contains(".o-dropdown-item:contains('Reply')", { count: 0 });
```

Comment:
This negative assertion doesn't match a positive assertion with exact same selector.
The 1st assertion used `.o-mail-Message-actions` for reply. I haven't run the code but wonder if this is also dropdown?

It's important for negative selectors to have exact same selector as positive assertion in same test, so that we have high confidence the negative assertion is correct. Negative assertions on their own is far too prone to passing when it shouldn't, i.e. typo in selector or selector was changed in template or label was changed. That's why it's important to have positive matching assertion so if it passes this one is relevant. If the positive assertion doesn't pass (was changed), then there's high chance we see this assertion too when changing the test and thus we update as needed.